### PR TITLE
[chore] - sparkle: enhance `Tree` component

### PR DESCRIPTION
**Pull request by JulesBelveze**
## Description

This PR enhances the Tree component in Sparkle to accommodate larger tree element sizes and improve its flexibility. The changes include:
- Added a new size prop to Tree.Item with options "sm" (default) and "md" for different padding sizes.
- Introduced an areActionsFading prop to control the visibility of action items.
- Updated the visual prop to accept a ComponentType instead of a ReactNode, allowing for better type checking and consistency.
- Improved the layout and styling of tree items, including better handling of icons, labels, and actions.
- Updated the stories to demonstrate the new features and provide examples of usage with various data sources.

All tributes to @Duncid I just tested it and fixed the breaking front issues in #6607 

## Risk

Breaking UX 

## Deploy Plan

- Deploy alpha sparkle and test any front breaking changes
- Deploy sparkle

### CODE REVIEW

**Comment by JulesBelveze** [comment]
These changes introduce a couple of breaking changes that are fixed in https://github.com/dust-tt/dust/pull/6607

File: sparkle/src/components/Tree.tsx [hunk 1]
```diff
@@ -49,13 +49,15 @@ interface TreeItemProps {
   label?: string;
   type?: "node" | "item" | "leaf";
   variant?: "file" | "folder" | "database" | "channel";
-  visual?: React.ReactNode;
+  size?: "sm" | "md";
+  visual?: ComponentType<{ className?: string }>;
```
Comment (fontanierh) [inline sparkle/src/components/Tree.tsx:53]: why this change ?

**Review by fontanierh — APPROVED** [review]
LGTM but curious about the reasoning behind the type change of the `visual` prop

File: sparkle/src/components/Tree.tsx (hunk 1 above)
Comment (JulesBelveze) [reply sparkle/src/components/Tree.tsx:53]: @fontanierh Yeah I don't know either, this is actually what breaks in https://github.com/dust-tt/dust/pull/6607.

Comment (JulesBelveze) [reply sparkle/src/components/Tree.tsx:53]: Dunno what was the rational behind but it could be to render all visuals as `Icon` and control the size?
